react/Viewer/helpers.js

```diff
@@ -8,12 +8,17 @@ import TextViewer from './TextViewer'
 import PdfMobileViewer from './PdfMobileViewer'
 import NoViewer from './NoViewer'
 import ShortcutViewer from './ShortcutViewer'
+import OnlyOfficeViewer from './OnlyOfficeViewer'
 
 export const isPlainText = (mimeType = '', fileName = '') => {
   return mimeType ? /^text\//.test(mimeType) : /\.(txt|md)$/.test(fileName)
 }
 
-export const getViewerComponentName = (file, isDesktop) => {
+export const getViewerComponentName = ({
+  file,
+  isDesktop,
+  isOnlyOfficeEnabled
+}) => {
```

Comment:
When converting this method to a React Component : 
- isDesktop should come from useBreakpoints() directly 
- I'm not fan of having to pass props from comp to comp. How about creating a context for the Viewer ?

Reply:
Hmmm moi je trouve que c'est assez clair de passer les props, au contraire on voit ce qui se passe et les dépendances. 

Ce qui n'est pas très clair je trouve c'est d'avoir `ViewerControl` comme wrapper qui passe `gestures, gesturesRef, onSwipe` au `ViewerByFile`

Ce qui donne le sentiment de passer les props de comp en comp c'est qu'on a un `ViewerByFile` intermédiaire... mais sinon en fait qu'on n'a que l'index puis le viewer, la ramification n'est pas très profonde en fait.

Reply:
> Hmmm moi je trouve que c'est assez clair de passer les props, au contraire on voit ce qui se passe et les dépendances.

When you start to need to pass props down on more than one level, it can be a sign of trouble. Specially if the first level doesn't need to know the props. We're back from the prop drilling, we should always keep that in mind ;)